ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,313 @@
+"""Collect insights and upload it to the Ingress service.
+
+It waits 5 min after startup before collects the data. Then it collects
+data after specified interval.
+
+When CP_OFFLINE_TOKEN is provided (either for prod or stage), it is used
+for ingress authentication instead of cluster pull-secret.
+"""
+
+import base64
+import io
+import json
+import logging
+import os
+import pathlib
+import sys
+import tarfile
+import time
+from typing import Any, Callable
+
+import kubernetes
+import requests
+
+OLS_USER_DATA_PATH = os.environ["OLS_USER_DATA_PATH"]
+OLS_USER_DATA_COLLECTION_INTERVAL = int(
+    os.environ.get("OLS_USER_DATA_COLLECTION_INTERVAL", 2 * 60 * 60)
+)  # 2 hours in seconds
+
+INITIAL_WAIT = 60 * 5  # 5 minutes in seconds
+RUN_WITHOUT_INITIAL_WAIT = bool(
+    os.environ.get("RUN_WITHOUT_INITIAL_WAIT", "false").lower() == "true"
+)
+
+INGRESS_ENV = os.environ.get("INGRESS_ENV", "stage")  # prod/stage
+if INGRESS_ENV not in {"prod", "stage"}:
+    raise ValueError(
+        f"Unknown value in INGRESS_ENV: {INGRESS_ENV}. Allowed: prod, stage"
+    )
+INGRESS_TIMEOUT = 10  # seconds
```

Comment:
seems short.  does this timeout include the time taken to actually upload the data, or just to make an initial connection?

Reply:
Don't know, but I raised it to 30.